Atividades/Em casa/atividade.js

```diff
@@ -0,0 +1,233 @@
+class HistoricoMedico {
+  constructor() {
+    this.consultas = [];
+    this.vacinas = [];
+    this.procedimento = [];
+  }
+}
+
+class Animal {
+  constructor(nome, idade, cor, consultas = []) {
+    this.nome = nome;
+    this.idade = idade;
+    this.cor = cor;
+    this.historico = new HistoricoMedico();
+  }
+
+  consultar() {
+    const dataConsulta = new Date();
+    this.consultas.push(dataConsulta);
+    this.historico.consultas.push(dataConsulta);
+  }
+
+  registrarProcedimento(procedimento) {
+    this.historico.procedimento.push(procedimento);
+  }
+
+  brincar() {
+    console.log(`${this.nome} está brincando.`);
+  }
+
+  acariciar() {
+    console.log(`${this.nome} está sendo acariciado.`);
+  }
+
+  alimentar() {
+    console.log(`${this.nome} está sendo alimentado.`);
+  }
+}
+
+class Gato extends Animal {
+  constructor(nome, idade, cor, consultas = []) {
+    super(nome, idade, cor, consultas);
+    this.social = true;
+  }
+
+  miar() {
+    if (this.social) {
+      console.log(`${this.nome} está ronronando.`);
+    } else {
+      console.log(`${this.nome} está sibilando.`);
+    }
+  }
+
+  vacinar(vacina) {
+    this.historico.vacinas.push(vacina);
+  }
+
+
+  brincar() {
+    console.log(`${this.nome} está brincando com uma bola de lã.`);
+  }
+
+  acariciar() {
+    super.acariciar();
+    if (this.social) {
+      console.log(`${this.nome} está ronronando de felicidade.`);
+    }
+  }
+
+  alimentar(comida) {
+    super.alimentar();
+    const comidasFavoritas = ["frango", "sachê", "peixe"];
+    let encontrouComidaFavorita = false;
+
+    for (let i = 0; i < comidasFavoritas.length; i++) {
+      if (comidasFavoritas[i] === comida) {
+        encontrouComidaFavorita = true;
+        break;
+      }
+    }
+
+    if (this.social && encontrouComidaFavorita) {
+      console.log(`${this.nome} está mais sociável agora.`);
+      this.social = true;
+    }
+  }
+}
+
+
+gato.miar(); // Saída: Nina está ronronando.
+gato.vacinar("gripe"); // Adiciona uma nova vacina e consulta
+gato.consultar(new Date()); // Adiciona uma nova consulta
+gato.brincar(); // Saída: Nina está brincando com uma bolinha de lã.
+gato.acariciar(); // Saída: Nina está sendo acariciado. Nina está ronronando de felicidade.
+gato.alimentar("frango"); // Saída: Nina está sendo alimentado. Nina está mais sociável agora.
+
+class Cachorro extends Animal {
+  constructor(nome, idade, cor, consultas = [], raça = []) {
+    super(nome, idade, cor, consultas);
+    this.raça = raça;
+    this.castrado = false;
+    this.ferido = false;
```

Comment:
acho legal ferido ser #ferido então ficaria: 
'''js
class Cachorro extends Animal {
   #feridp
  constructor(nome, idade, cor, consultas = [], raça = []) {
    super(nome, idade, cor, consultas);
    this.raça = raça;
    this.castrado = false;
    this.#ferido = false;
  }
```
usaria assim pq # faz com que as pessoas nao possam acesar com gato.ferido e mudar para true assim fazendo com que só sua função mude